docs/proposals/perf.md

```diff
@@ -103,9 +108,73 @@ This VM is used to run KubeEdge Cloud Part Services including Edge Controller an
 These VMs are used to deploy numbers of KubeEdge Edge Nodes pods which are running Edged and EdgeHub and so on.
 We will adjust the Count of VMs based on the KubeEdge Edge Nodes numbers.
 
+KubeEdge Performance Test setup is similar with K8S KubeMark setup,
+where they simulate numbers of hollow-nodes in pods and deploy on K8S Cluster.
+In KubeEdge we also do the similar kind of simulation for creating KubeEdge Edge Nodes
+in pods and deploy the pods through deployment. Our pod takes up resources as below:
+- 1 pod : 0.10 vCPU & 250MB RAM
+
+With KubeEdge pod deployment we can accomodate 10 pods/1vCPU approximately.
+Base on the above K8S Nodes flavor, the CPU and RAM are 32vCPU and 128GB respectively.
+Per K8S Node we should be able to deploy 320 pods(KubeEdge Edge Nodes)/32vCPU
+and RAM consumption would be around 80GB. If we have 5 K8S Nodes with the similar flavor,
+on a whole we should be able to deploy 1500 pods(KubeEdge Edge Nodes)/5 K8S Nodes.
+
 ### Performance Test Framework
 <img src="../images/perf/perf-test-framework.png">
 
+KubeEdge Performance Test Framework will be designed based on the **Gomega** and **Ginkgo**.
+
+The Performance Test Framework mainly relates to Utils Library and different types of tests:
+- E2E Test
+- Latency Test
+- Load Test
+- Scalability Test
+- ...
+
+E2E Test Sample:
```

Comment:
This is a great test case. Can you help me understand how you measure the time duration for latency and throughput?

Reply:
@CindyXing , Above is sample test case given deploy application to kubeeedge node and check the pods are coming into ready state, above sample case doesnt have measurement criteria for pod startup time, that logic of measuring the latency yet be added.

while doing measurement from framework, we will take the pod created time(t1) stamp from test cases, and get the timestamp when pod gets into the running state(t2), from the t1 and t2, we will calculate the time taken for pod to come up and Running.

Coming to throughput, From the tests we do the parallel depployments using go-routunes (Ginkgo has build-in support to go routines) on single-node and calculate the CPU,Memory consupmtion for diffrent no.of deployments done.

Reply:
What if I want the specific latency for cloudhub/edgehub message delivery? Measuring e2e is a blackbox validation of the perf. Here we should do whitebox. That's how we could quickly identify any potential issue(s). Thanks

Reply:
Thanks @CindyXing. This scenario is added in the latest commit.